app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,21 @@
+class LikesController < ApplicationController
+
+  before_action :authenticate_user!, :except => [:index]
+
+  def new
+    @post = Post.find(params[:post_id])
+    @like = Like.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    if current_user.liked_post? @post
+       redirect_to posts_path
+    else
+      new_like = @post.likes.create(:user_id => current_user)
```

Comment:
Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)